src/kernel/clock.c

```diff
@@ -51,6 +72,16 @@ closure_function(1, 1, boolean, timer_adjust_handler,
     return true;
 }
 
+void clock_rtc_step(s64 step)
+{
+    vdso_update_gen();
+    __vdso_dat->rtc_offset += step;
+    vdso_update_gen();
+    pqueue_walk(kernel_timers->pq, stack_closure(timer_adjust_handler, step));
```

Comment:
It seems to me that this pqueue_walk() - and the one in clock_reset_rtc() - should be called with the timerqueue lock held. Maybe a timer_walk() function could be added to timer.c that does this?

Reply:
Yes, that should be protected. I was thinking that when Francesco merges his PR for the timer changes I update this to use that solution since he's already solved it.

Reply:
I updated clock_step_rtc() to use the new timer_adjust functions to protect the timer updates.